lib/util/files/finder.js

```diff
@@ -1,59 +1,40 @@
-let { readdir, stat } = require("fs/promises");
-let path = require("path");
+let { readdir, stat } = require("node:fs/promises");
+let path = require("node:path");
 
 exports.FileFinder = class FileFinder {
-	constructor(directory, { skipDotfiles, filter = () => true } = {}) {
+	constructor(directory, { skipDotfiles, filter } = {}) {
 		this.directory = directory;
 		this.filter = filename => {
-			if(skipDotfiles && isDotfile(filename)) {
+			if(skipDotfiles && path.basename(filename).startsWith(".")) {
 				return false;
 			}
-			return filter(filename);
+			return filter === undefined ? true : filter(filename);
 		};
 	}
 
 	// returns a list of relative file paths within the respective directory
-	all() {
-		return tree(this.directory).
-			then(filenames => filenames.filter(this.filter));
+	async all() {
+		let filenames = await tree(this.directory);
+		return filenames.filter(this.filter);
 	}
 
 	// returns all file paths that match the filter function
-	match(filepaths) {
-		return filesWithinDirectory(this.directory, filepaths).
-			then(filepaths => filepaths.filter(this.filter));
+	async match(filepaths) {
```

Comment:
`async` isn't actually required by the implementation nor by downstream (i.e. faucet-static and faucet-images), but current tests demand it...

... except if the `filter` option is async, so I guess it's best to leave this as is, if only because it allows for more flexibility.